### PR TITLE
WHO 11 cards (22/10)

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/mondassian_colony_ship.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:Mondassian Colony Ship
+ManaCost:no cost
+Types:Plane Spacecraft
+T:Mode$ Attacks | ValidCard$ Creature | TriggerZones$ Command | Execute$ AnimosityPump | TriggerDescription$ Whenever a creature attacks, it gets +1/+1 until end of turn for each other creature its controller controls that shares a creature type with it.
+SVar:AnimosityPump:DB$ Pump | Defined$ TriggeredAttackerLKICopy | NumAtt$ X | NumDef$ X
+SVar:X:Count$Valid Creature.attacking+NotTriggeredAttacker+sharesCreatureTypeWith TriggeredAttacker
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/mondassian_colony_ship.txt:6]: wrong valid

File: forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Name:Nanogene Conversion
+ManaCost:3 U
+Types:Sorcery
+A:SP$ Pump | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Choose target creature you control | RememberObjects$ Targeted | SubAbility$ DBRepeatEach | StackDescription$ Each other creature {p:You} controls becomes a copy of {c:Targeted} until end of turn, except those creatures aren't legendary if {c:Targeted} is legendary. | SpellDescription$ Choose target creature you control. Each other creature becomes a copy of it until end of turn, except those creatures aren't legendary if the chosen creature is legendary.
+SVar:DBRepeatEach:DB$ RepeatEach | RepeatCards$ Creature.IsNotRemembered | UseImprinted$ True | RepeatSubAbility$ DBClone | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt:5]: use `NotDefinedTargeted` for cleaner script

File: forge-gui/res/cardsfolder/upcoming/new_new_york.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Name:New New York
+ManaCost:no cost
+Types:Plane New Earth
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | TriggerZones$ Command | Execute$ TrigAnimate | TriggerDescription$ At the beginning of combat on your turn, until end of turn, noncreature artifacts you control become 3/3 Vehicles in addition to their other types and gain flying, haste, and crew 1.
+SVar:TrigAnimate:DB$ AnimateAll | ValidCards$ Artifact.nonCreature+YouCtrl | Types$ Vehicle | Power$ 3 | Toughness$ 3 | Keywords$ Flying & Haste & Crew:1
+T:Mode$ ChaosEnsues | TriggerZones$ Command | Execute$ TrigToken | TriggerDescription$ Whenever chaos ensues, create a Treasure token and a 2/2 white Alien creature token.
+SVar:TrigToken:DB$ Token | TokenAmount$ X | TokenScript$ c_a_treasure_sac,w_2_2_alien | TokenOwner$ You
+DeckHas:Type$Alien|Token & Ability$Token
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/new_new_york.txt:8]: invalid type

File: forge-gui/res/cardsfolder/upcoming/ominous_cemetery.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Name:Ominous Cemetery
+ManaCost:no cost
+Types:Land
+A:AB$ Mana | Cost$ T | Produced$ C | Amount$ 1 | SpellDescription$ Add {C}
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/ominous_cemetery.txt:4]: 3 . missing

File: forge-gui/res/cardsfolder/upcoming/rms_titanic.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+PT:7/1
+K:Flying
+K:Trample
+T:Mode$ DamageDone | ValidSource$ Card.Self | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigSac | TriggerDescription$ When CARDNAME deals combat damage to a player, sacrifice it and create that many Treasure tokens.
+SVar:TrigSac:DB$ Sacrifice | SacValid$ Self | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenAmount$ X | TokenScript$ c_a_treasure_sac | TokenOwner$ You
+SVar:X:TriggerCount$DamageAmount
+K:Crew:3
+DeckHas:Ability$Sacrifice|Token & Type$Treasure
+Oracle:Flying\nWard {2} (Whenever this creature becomes the target of a spell or ability an opponent controls, counter it unless that player pays {2}.)\nWhen Archive Dragon enters the battlefield, scry 2.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rms_titanic.txt:13]: ?

File: forge-gui/res/cardsfolder/upcoming/the_cyber_controller.txt [hunk 6]
```diff
@@ -0,0 +1,13 @@
+Name:The Cyber-Controller
+ManaCost:X U U B
+Types:Legendary Artifact Creature Cyberman
+PT:3/3
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigMill1 | TriggerDescription$ When CARDNAME enters the battlefield, each opponent mills X cards. Put all creature cards milled this way onto the battlefield face down under your control. They're 2/2 Cyberman artifact creatures.
+SVar:TrigMill1:DB$ Mill | Defined$ Opponent | NumCards$ X | RememberMilled$ True | SubAbility$ DBChangeZoneAll
+SVar:DBChangeZoneAll:DB$ ChangeZoneAll | Origin$ Graveyard | GainControl$ True | Destination$ Battlefield | ChangeType$ Card.Creature+IsRemembered | FaceDown$ True | FaceDownPower$ 2 | FaceDownToughness$ 2 | FaceDownSetType$ Artifact & Creature & Cyberman | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_cyber_controller.txt:7]: add Exile

File: forge-gui/res/cardsfolder/upcoming/zygon_infiltrator.txt [hunk 7]
```diff
@@ -0,0 +1,8 @@
+Name:Zygon Infiltrator
+ManaCost:2 U
+Types:Creature Alien Shapeshifter Soldier
+PT:3/2
+A:AB$ Tap | Cost$ 2 U | PrecostDesc$ Body-print - | SorcerySpeed$ True | ValidTgts$ Creature.Other | TgtPrompt$ Select another target creature | SubAbility$ DBCounter | SpellDescription$ Tap another target creature and put a stun counter on it. Zygon Infiltrator becomes a copy of that creature for as long as that creature remains tapped. Activate only as a sorcery. (If a permanent with a stun counter would become untapped, remove one from it instead.)
+SVar:DBCounter:DB$ PutCounter | Defined$ Targeted | CounterType$ Stun | RememberCounterNum$ 1 | SubAbility$ DBCopy
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/zygon_infiltrator.txt:6]: why remember

File: forge-gui/res/cardsfolder/upcoming/mondassian_colony_ship.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/mondassian_colony_ship.txt]: Any advice for the valid ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/mondassian_colony_ship.txt]: it doesn't work now?

File: forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Name:Nanogene Conversion
+ManaCost:3 U
+Types:Sorcery
+A:SP$ Pump | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Choose target creature you control | SubAbility$ DBRepeatEach | StackDescription$ Each other creature {p:You} controls becomes a copy of {c:Targeted} until end of turn, except those creatures aren't legendary if {c:Targeted} is legendary. | SpellDescription$ Choose target creature you control. Each other creature becomes a copy of it until end of turn, except those creatures aren't legendary if the chosen creature is legendary.
+SVar:DBRepeatEach:DB$ RepeatEach | RepeatCards$ Creature.NotDefinedTargeted | RepeatSubAbility$ DBClone | SubAbility$ DBCleanup
+SVar:DBClone:DB$ Clone | Defined$ Targeted | CloneTarget$ Remembered | NonLegendary$ True | Duration$ UntilEndOfTurn
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt:7]: remembered now obsolete

File: forge-gui/res/cardsfolder/upcoming/mondassian_colony_ship.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/mondassian_colony_ship.txt]: This version works now

File: forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt [hunk 9]
```diff
@@ -0,0 +1,7 @@
+Name:Nanogene Conversion
+ManaCost:3 U
+Types:Sorcery
+A:SP$ Pump | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Choose target creature you control | SubAbility$ DBRepeatEach | StackDescription$ Each other creature {p:You} controls becomes a copy of {c:Targeted} until end of turn, except those creatures aren't legendary if {c:Targeted} is legendary. | SpellDescription$ Choose target creature you control. Each other creature becomes a copy of it until end of turn, except those creatures aren't legendary if the chosen creature is legendary.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt:4]: wrong SpellDescription

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt]: You should be mindful of StackDescriptions too:
image

File: forge-gui/res/cardsfolder/upcoming/the_sea_devils.txt [hunk 10]
```diff
@@ -0,0 +1,11 @@
+Name:The Sea Devils
+ManaCost:2 G
+Types:Enchantment Saga
+K:Saga:3:DBToken,DBToken,DBEffect
+SVar:DBToken:DB$ Token | TokenScript$ g_2_2_alien_salamander_islandwalk | SpellDescription$ Create a 2/2 green Alien Salamander creature token with islandwalk. (It can't be blocked as long as defending player controls an Island.)
+SVar:DBEffect:DB$ Effect | Triggers$ TrigDamage | SubAbility$ DBCleanup | SpellDescription$ Until end of turn, whenever a Salamander deals combat damage to a player, it deals that much damage to target creature that player controls.
+SVar:TrigDamage:Mode$ DamageDone | ValidSource$ Salamander.YouCtrl | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigDealDamage | TriggerDescription$ Whenever a Salamander deals combat damage to a player, it deals that much damage to target creature that player controls.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_sea_devils.txt:7]: wrong valid

File: forge-gui/res/cardsfolder/upcoming/the_sea_devils.txt [hunk 11]
```diff
@@ -0,0 +1,11 @@
+Name:The Sea Devils
+ManaCost:2 G
+Types:Enchantment Saga
+K:Saga:3:DBToken,DBToken,DBEffect
+SVar:DBToken:DB$ Token | TokenScript$ g_2_2_alien_salamander_islandwalk | SpellDescription$ Create a 2/2 green Alien Salamander creature token with islandwalk. (It can't be blocked as long as defending player controls an Island.)
+SVar:DBEffect:DB$ Effect | Triggers$ TrigDamage | SubAbility$ DBCleanup | SpellDescription$ Until end of turn, whenever a Salamander deals combat damage to a player, it deals that much damage to target creature that player controls.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_sea_devils.txt]: no sub

File: forge-gui/res/cardsfolder/upcoming/the_sea_devils.txt [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+SVar:TrigDamage:Mode$ DamageDone | ValidSource$ Salamander.YouCtrl | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigDealDamage | TriggerDescription$ Whenever a Salamander deals combat damage to a player, it deals that much damage to target creature that player controls.
+SVar:TrigDealDamage:DB$ DealDamage | ValidTgts$ Creature.ControlledBy TriggeredTarget | NumDmg$ X | TgtPrompt$ Select target creature that player controls
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_sea_devils.txt:8]: needs DamageSource$

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_sea_devils.txt]: TriggeredCard does not exist...

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_sea_devils.txt]: TriggeredSource  ?

File: forge-gui/res/cardsfolder/upcoming/the_cyber_controller.txt [hunk 13]
```diff
@@ -0,0 +1,13 @@
+Name:The Cyber-Controller
+ManaCost:X U U B
+Types:Legendary Artifact Creature Cyberman
+PT:3/3
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigMill1 | TriggerDescription$ When CARDNAME enters the battlefield, each opponent mills X cards. Put all creature cards milled this way onto the battlefield face down under your control. They're 2/2 Cyberman artifact creatures.
+SVar:TrigMill1:DB$ Mill | Defined$ Opponent | NumCards$ X | RememberMilled$ True | SubAbility$ DBChangeZoneAll
+SVar:DBChangeZoneAll:DB$ ChangeZoneAll | Origin$ Graveyard,Exile | GainControl$ True | Destination$ Battlefield | ChangeType$ Card.Creature+IsRemembered | FaceDown$ True | FaceDownPower$ 2 | FaceDownToughness$ 2 | FaceDownSetType$ Artifact & Creature & Cyberman | SubAbility$ DBCleanup
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/upcoming/the_cyber_controller.txt:7]: @tool4ever shouldn't be `Defined$ Remembered` be better?

Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/the_cyber_controller.txt:7]: I think so... would need to be `DB$ ChangeZone` instead though

File: forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt [hunk 14]
```diff
@@ -0,0 +1,7 @@
+Name:Nanogene Conversion
+ManaCost:3 U
+Types:Sorcery
+A:SP$ Pump | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Choose target creature you control | SubAbility$ DBRepeatEach | StackDescription$ Each other creature becomes a copy of {c:Targeted} until end of turn, except those creatures aren't legendary | SpellDescription$ Choose target creature you control. Each other creature becomes a copy of that creature until end of turn, except it isn't legendary.
+SVar:DBRepeatEach:DB$ RepeatEach | RepeatCards$ Creature.NotDefinedTargeted | RepeatSubAbility$ DBClone
+SVar:DBClone:DB$ Clone | Defined$ Targeted | CloneTarget$ Remembered | NonLegendary$ True | Duration$ UntilEndOfTurn
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt:6]: Probably should add loop functionality to CloneEffect so we don't need repeats for cards like this

File: forge-gui/res/cardsfolder/upcoming/renegade_silent.txt [hunk 15]
```diff
@@ -0,0 +1,10 @@
+Name:Renegade Silent
+ManaCost:3 U
+Types:Creature Alien Horror
+PT:3/3
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigGoad | TriggerDescription$ At the beginning of your end step, goad up to one target creature you don't control and put a +1/+1 counter on CARDNAME. CARDNAME phases out. (Treat it and anything attached to it as though they don't exist until your next turn.)
+SVar:TrigGoad:DB$ Goad | ValidTgts$ Creature.YouDontCtrl | TargetMin$ 0 | TargetMax$ 1 | SubAbility$ DBPutCounter
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/renegade_silent.txt:6]: You'll need TgtPrompt for this

File: forge-gui/res/cardsfolder/upcoming/return_the_past.txt [hunk 16]
```diff
@@ -0,0 +1,7 @@
+Name:Return the Past
+ManaCost:4 R R
+Types:Enchantment
+S:Mode$ Continuous | Affected$ Instant.YouOwn,Sorcery.YouOwn | AffectedZone$ Graveyard | AddKeyword$ Flashback | Condition$ PlayerTurn | Description$ As long as it's your turn, each instant and sorcery card in your graveyard has flashback. The flashback cost is equal to that card's mana cost.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/return_the_past.txt:4]: Last sentence in Desc is different than oracle... which is correct?

**Review by tool4ever — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt (hunk 14 above)
Comment (kevlahnota) [reply forge-gui/res/cardsfolder/upcoming/nanogene_conversion.txt:6]: gonna merge this and maybe add the new functionality on new MR

**Pull request merged**